src/github.com/couchbase/sync_gateway/rest/changes_api.go

```diff
@@ -265,6 +293,143 @@ func (h *handler) sendSimpleChanges(channels base.Set, options db.ChangesOptions
 	return nil, forceClose
 }
 
+/*
+ * Generate the changes for a specific list of doc ID's, only documents accesible to the user will generate
+ * results
+ */
+func (h *handler) sendChangesForDocIds(userChannels base.Set, explicitDocIds []string, options db.ChangesOptions) (error, bool) {
+
+	// Get the set of channels the user has access to; nil if user is admin or has access to user "*"
+	var availableChannels channels.TimedSet
+	if h.user != nil {
+		availableChannels = h.user.InheritedChannels()
+		if availableChannels == nil {
+			panic("no channels for user?")
+		}
+		if availableChannels.Contains(channels.UserStarChannel) {
+			availableChannels = nil
+		}
+	}
+
+	filterChannelSet := func(channelMap channels.ChannelMap) []string {
+		var result []string
+		if availableChannels == nil {
+			result = []string{}
+		}
+		for ch, rm := range channelMap {
+			if availableChannels == nil || availableChannels.Contains(ch) {
+				//Do not include channels doc removed from in this rev
+				if rm == nil {
+					result = append(result, ch)
+				}
+			}
+		}
+		return result
+	}
+
+	type changesRowValue struct {
+		Rev      string              `json:"rev"`
+		Channels []string            `json:"channels,omitempty"`
+		Access   map[string]base.Set `json:"access,omitempty"` // for admins only
+	}
+	type changesRow struct {
+		UpdateSeq uint64           `json:"seq,omitempty"`
+		ID        string           `json:"id,omitempty"`
+		Doc       db.Body          `json:"doc,omitempty"`
+		Changes   []*changesRowValue `json:"changes,omitempty"`
+		Error     string           `json:"error,omitempty"`
+		Status    int              `json:"status,omitempty"`
+	}
+
+	// Subroutine that creates a response row for a document:
+	first := true
+	var lastSeq uint64 = 0
+	rowMap := make(map[uint64]*changesRow)
+
+	createRow := func(doc db.IDAndRev) *changesRow {
+		row := &changesRow{ID: doc.DocID}
+		value := changesRowValue{}
+
+		// Fetch the document body and other metadata that lives with it:
+		body, channelSet, _, _, sequence, err := h.db.GetRevAndChannels(doc.DocID, doc.RevID, false)
+		if err != nil {
+			return nil
```

Comment:
Is this error expected?  If not - should we either be returning an error response, or logging something?